src/Streaming/Internal.hs

```diff
@@ -122,9 +123,10 @@ import Data.IORef
     \- or, in the producer case, 'Streaming.Prelude.next'
     The constructors are exported by the 'Internal' module.
 -}
-data Stream f m r = Step !(f (Stream f m r))
+data Stream f m r = forall x. Step !(f x) (x -> Stream f m r)
                   | Effect (m (Stream f m r))
                   | Return r
+{-
```

Comment:
One downside is that the opacity of the existential now means we can't automatically generate `Show`, `Eq` and `Data`.